src/components/item/item-sliding.ts

```diff
@@ -47,19 +81,154 @@ import {List} from '../list/list';
+
+  @ContentChild(Item) item;
 
-  constructor(@Optional() private _list: List, elementRef: ElementRef) {
+  constructor( @Optional() private _list: List, private _renderer: Renderer, private _elementRef: ElementRef) {
     _list.enableSlidingItems(true);
-    elementRef.nativeElement.$ionSlide = ++slideIds;
+    _elementRef.nativeElement.$ionComponent = this;
+  }
+
+  /**
+   * @private
+   */
+  @ContentChildren(ItemOptions)
+  set _itemOptions(itemOptions) {
+    var sides = 0;
+    for (var item of itemOptions.toArray()) {
+      var side = item.getSides();
+      switch(side) {
+        case SideFlags.Left: this._leftOptions = item; break;
+        case SideFlags.Right: this._rightOptions = item; break;
+      }
+      sides |= item.getSides();
+    }
+    this._optsDirty = true;
+    this._sides = sides;
+  }
+
+  /**
+   * @private
+   */
+  _startSliding(startX: number) {
+    this._addClass('active-slide');
+    this._offsetX = this._openAmount;
+    this._startX = startX;
+  }
+
+  /**
+   * @private
+   */
+  _moveSliding(x: number): number {
+    if(this._optsDirty) {
+      if(this._optsWidthRightSide == 0 && this._rightOptions) {
+        this._optsWidthRightSide = this._rightOptions.getWidth();
+      }
+      if(this._leftOptions) {
+        this._optsWidthLeftSide = this._leftOptions.getWidth();
+      }
+      this._optsDirty = false;
+    }
+
+    this._openAmount = computeOpenAmount(
+      x,
+      this._startX,
+      this._offsetX,
+      this._optsWidthRightSide,
+      this._optsWidthLeftSide,
+      this._sides
+    );
+    this._updateVisual(false);
+    return this._openAmount;
+  }
+
+  /**
+   * @private
+   */
+  _endSliding(velocity: number): number {
+    this._openAmount = computeRestingPoint(
+      this._openAmount,
+      this._optsWidthRightSide,
+      this._optsWidthLeftSide,
+      velocity
+    );
+    this._updateVisual(true);
+    return this._openAmount;
+  }
+
+  /**
+   * @private
+   */
+  private _updateVisual(isFinal: boolean) {
+    if(this._timer) {
+      clearTimeout(this._timer);
+      this._timer = null;
+    }
+    let openAmount = this._openAmount;
+    let didEnd = openAmount == 0;
+    if(didEnd) {
+        // TODO: refactor. there must exist a better way
+        // if sliding ended, we wait 400ms until animation finishes
+        this._timer = setTimeout(()=> {
+          this._removeClass('active-slide');
+          this._removeClass('active-options-right');
+          this._removeClass('active-options-left');
+          this._timer = null;
+        }, 400);
+
+    }else if(openAmount > 0) {
+        this._addClass('active-options-right');
+    }else if(openAmount < 0) {
+        this._addClass('active-options-left');
+    }
+
+    this.item.setCssStyle(CSS.transition, (isFinal ? '' : 'none'));
+    this.item.setCssStyle(CSS.transform, (didEnd ? '' : 'translate3d(' + -openAmount + 'px,0,0)'));
+  }
+
+  /**
+   * @private
+   */
+  private _addClass(className: string) {
```

Comment:
_addClass and _removeClass could be combined like how the renderer's `setElementClass` works, where the 3rd arg is a boolean. Instead you could just have `_setClass`